### DISCUSSION TITLE
How to create an observer for sum of 'Total fraction of dose'

### DISCUSSION THREAD

**Discussion by arketkar** [Open-Systems-Pharmacology/Forum#770] [Questions & Problems]
I have a case where the drug is metabolized by three CYP enzymes to form a single metabolite. I have data for metabolite fraction as well. However in the current set-up the 'Total fraction of dose' gives output for fractions metabolized by three enzymes separately. I require sum of these fractions.

Appreciate if someone guides me on creating an output/observer to determine the sum of these three fractions, which then could be optimized against the reported value of fraction metabolized.

Thanks.
Anant

**Comment by StephanSchaller** [comment]
Hi Anant,

three CYPs for the same metabolite? Do you know how much each enzyme contributes to the total fraction of the three? 
Then you could optimize with three fraction values.

Are other metabolism/clearance processes involved?
Do you have metabolite time-concentration data?

**Comment by arketkar** [comment]
Hi @StephanSchaller 

Yes it is reported that all the three CYPs contribute to the same metabolite. I do not know individual enzyme contributions, however I have Vmax and Km values generated with individual rCYPs, and total microsomal Vmax and Km values.

There is one UGT pathway which forms one metabolite and the 3rd major clearance is urinary excretion.

I have metabolite urinary time-excretion data.

**Comment by StephanSchaller** [comment]
So how did extrapolation from in-vitro work out? Did you try scaling rCYP or microsomal CL?
It sounds like there is quite some info/data available and going straight into fitting might not be the best idea.

But if you want to create an Observer, just export your simulation to MoBi. Build & test the Observer, then save the Observer and re-import it to PK-Sim.

An explanation how to create an observer can be found [here](https://docs.open-systems-pharmacology.org/working-with-mobi/mobi-documentation/model-building-components#observers)

Best,
Stephan

**Comment by arketkar** [comment]
@StephanSchaller 
I tried rCYP data, and it is reasonably working with some optimization of Vmax values.

Yes there is quite some data on this compound, and as rightly mentioned by you based on thorough literature search on metabolic pathways of active and metabolites I have initiated the fitting process.

Thanks for the inputs provided on creating an Observer.

Since I have not a good hold on MoBi functionalities, are there any tips to utilize the existing observers of the 'total fraction of dose' for individual enzyme and link them in MoBi?

Is there any upcoming on-line workshop on MoBi or any recordings of demonstrations with real-time case studies?

Thanks and appreciate your inputs.
Anant

**Comment by moPellowe** [comment]
I am trying to do something similar. I am trying to create an observer that will sum up two metabolites in the whole body. I created an observer and am trying to use a sum formula (just like in the Total fraction of dose observer). I set the molecule to the metabolites of interest, set the match tag condition to MOLECULE and left the sum formula as it is by default (since I just want to add up the amounts). This gives me the total amount of metabolite for each of them indvidually, but how do I get it to add two molecules together into the same value? I tried adding a reference to the other metabolite but MoBi cannot resolve the path.

![image](https://user-images.githubusercontent.com/57540473/177562705-1f68bdf5-596b-44e0-a494-2e779f3b2077.png)

**Reply by msevestre** [reply]
You will need to make this variable dynamic as well by adding _#i.

**Comment by Yuri05** [comment]
There are several ways to do this. One possibility: use the new OR operator for container criteria in MoBi 11.
In the attached project, I have a compound **C1** and 3 metabolites.
[P1.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/9056144/P1.zip)

I defined compound specific parameter **Total amount metabolized** which sums up all 3 metabolites:
![grafik](https://user-images.githubusercontent.com/25061876/177581386-b08b9e05-7839-4889-93c9-954d17a7da7c.png)

<details><summary><b>In a simulation, this will be expanded to:</b></summary>

![grafik](https://user-images.githubusercontent.com/25061876/177583668-84e71184-1e13-4b83-8467-1cd99679b5b8.png)
</details>

Then I defined the total fraction metabolized as `Sum_of_all_metabolites_amounts / Total_drugmass`:
![grafik](https://user-images.githubusercontent.com/25061876/177588115-32d51587-c551-4c82-8d52-89823a1fab87.png)

(here I defined it as a **global compound parameter**, which I then set plottable. Alternatively, I could define an _observer_ instead).

![grafik](https://user-images.githubusercontent.com/25061876/177586657-4cd58ec9-dfd3-4fb8-874c-dbac095388e3.png)

Another way to do this (and that's what Michael mentioned above) is in this project: [P2.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/9056366/P2.zip)

Here you sum up only the first metabolite via conditions, and in the formula you add all other metabolites. 
The important point is adding `_#i` to all other metabolites (otherwise MoBi tries to locate them not relative to the first metabolite but relative to the parameter (or in you case: to the observer), which cannot work.
Please also check the explanation of how sum formulas are working (s. screenshot)
![grafik](https://user-images.githubusercontent.com/25061876/177595867-a80938c1-d05f-4707-8fe1-9b57e7044365.png)

<details><summary><b>In a simulation, this will be expanded to:</b></summary>
Same formula as in the 1st approach, just written differently:

![grafik](https://user-images.githubusercontent.com/25061876/177596738-b7f1137c-589e-4205-ad89-4c5d77897af3.png)
</details>

=================================================================================
In general, the first approach is more flexible. If you have a container where e.g. the first metabolite exists, but the 2nd metabolite does not - the sum formula from the 2nd approach would crash, because it explicitly looks for all metabolites in every container where the 1st metabolite is present
Such a situation is never the case for models exported from PK-Sim, but theoretically it could happen if you extend your model in MoBi.

**Reply by moPellowe** [reply]
Thanks so much for this thorough answer, @Yuri05 .
